students/ZackConnaughton/lessons/lesson10/assignment/database.py

```diff
@@ -0,0 +1,183 @@
+"""
+creates database with customer info and product info
+"""
+
+import csv
+from pymongo import MongoClient
+import time
+import inspect
+
+def timing(func):
+    def wrapper(*args, **kwargs):
+        output = [func.__name__]
+        collection = ""
+        arguments = (inspect.getfullargspec(func))
+        if [arg for arg in arguments if arg == ['collection']]:
+            collection = args[0]
+            start_count = (data_count(collection))
+        start = time.clock()
+        func(*args, **kwargs)
+        output.append(time.clock() - start)
+        if collection:
+            end_count = data_count(collection)
+            output.append(abs(end_count - start_count))
+        else:
+            output.append("N/A")
+        csv_output(output)
+    return wrapper
```

Comment:
This one should definitely have a doc string. :) Nice use of inspect. It does look like some of the functions that process records do not have collection in their arguments, though. Lol, just saw the file below this one is the one I am suppose to review. So, maybe ignore this?